rustls/src/msgs/handshake.rs

```diff
@@ -220,8 +220,11 @@ impl TlsListElement for NamedGroup {
     };
 }
 
+/// RFC8446: `SignatureScheme supported_signature_algorithms<2..2^16-2>;`
 impl TlsListElement for SignatureScheme {
-    const SIZE_LEN: ListLength = ListLength::U16;
+    const SIZE_LEN: ListLength = ListLength::NonZeroU16 {
```

Comment:
Should we generalize to `ListLength::U16Min` so we can enforce the `2..`?

Reply:
The `2..` is enforced, but via a different route: if the length is non-zero, then we read the items (each 2 bytes in length) until the end is reached. If the length is 1 (or any odd value) then the last item read will fail, with `InvalidMessage::MissingData`.